### PR TITLE
Add fail-on [TOOL-1217]

### PR DESCRIPTION
**Pull request by leifdreizler**
Stole what I could from your BK plugin @eculver. My bash is just ok so have some questions which are inline in the code.

Goals for the review should be making sure things are backwards compatible and that support for the various flags work. Right now the default behavior is to show all vulns, upload them to Snyk, but never to fail.

### CODE REVIEW

File: initialization/snyk.sh [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
 
-./snyk-linux test -d || true # will always pass, but still send results up to Snyk                                                                                                
-./snyk-linux monitor -d || true # suppresses errors w/ snyk monitor (which shouldn't have any)                                                                                    
+echo "~~~ Running Snyk tests"
+./snyk test ${flags[@]}
 
-if [ "$SNYK_LEVEL" = "FLHI" ]; then
-    ./snyk-linux test --severity-threshold=high -q
+if [[ "${NEVER_FAIL}" = true ]]; then
+  exit 0;
```
Comment (leifdreizler) [inline initialization/snyk.sh]: This is based on https://github.com/segmentio/snyk-buildkite-plugin/blob/master/hooks/command#L73-L75

I thought it needed to be `true`, not `"true"` and that `set +e` would just maintain the existing failure code, not guarantee the code would be 0 so made these changes. Does the BK version need to get updated?

File: initialization/snyk.sh [hunk 2]
```diff
@@ -1,19 +1,45 @@
...
+flags=(
+  "--severity-threshold=$severity_threshold"
+  "--fail-on=$fail_on"
+  "--org=$org"
+)
+
+monitor_flags=(
+  "--org=$org"
+)
+
```
Comment (leifdreizler) [inline initialization/snyk.sh]: I didn't really know what the environment stuff was doing, do I need that? 

https://github.com/segmentio/snyk-buildkite-plugin/blob/master/hooks/command#L60-L63

**Comment by eculver** [comment]
I think this is pretty much ready. I will do some more final testing tomorrow, but this seems to work and is backwards compatible.

File: initialization/snyk.sh [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
-[ ! -f snyk-linux ] && { echo "Snyk failed to download!"; exit 0; }
+# "never" is not a valid input, but we make it a valid input to this script, so we
+# need to swap it out so that the CLI doesn't complain about "never" not being a thing
+if [ "${fail_on}" = "never" ]; then
+  fail_on=all
+  NEVER_FAIL="true"
+fi
+
+# prevent the script from ever exiting non-zero
+if [ "${NEVER_FAIL}" = "true" ]; then
```
Comment (leifdreizler) [inline initialization/snyk.sh]: @eculver should this block get removed?

**Review by eculver — APPROVED** [review]

**Pull request merged**